File: dex-tracker-frontend/src/types/pokedex.ts
export type Game = {
  name: string;
  displayName: string;
};

export type PokedexType = "NATIONAL" | "REGIONAL";

export type Pokedex = {
  name: string;
  displayName: string;
  region: string;
  type: PokedexType;
  gen: number;
  entries: Array<DexEntry>;
};

export type DexEntry = {
  name: string;
  number: number;
};

File: dex-tracker-backend/src/main/kotlin/com/github/aleperaltabazas/dex/model/Pokedex.kt
package com.github.aleperaltabazas.dex.model

enum class PokedexType {
    NATIONAL,
    REGIONAL,
}

data class Pokedex(
    val name: String,
    val displayName: String,
    val region: String,
    val gen: Int,
    val type: PokedexType,
    val entries: List<Entry>,
)

data class Entry(
    val name: String,
    val number: Int,
)

data class Game(
    val name: String,
    val displayName: String,
    val gen: Int,
) {
    constructor(pokedex: Pokedex) : this(
        name = pokedex.name,
        displayName = pokedex.displayName,
        gen = pokedex.gen,
    )
}
